internal/servers/controller/handlers/targets/target_service.go

```diff
@@ -1104,6 +1104,28 @@ func (s Service) AuthorizeSession(ctx context.Context, req *pbs.AuthorizeSession
 					return nil, errors.Wrap(ctx, err, op, errors.WithMsg("creating proto struct for secret"))
 				}
 			}
+
+			var credType string
+			var credData *structpb.Struct
+			if l.CredentialType() != credential.UnspecifiedType {
+				credType = string(l.CredentialType())
+				data := make(map[string]interface{})
+
+				switch c := cred.(type) {
+				case credential.UserPassword:
+					data["username"] = c.Username()
+					data["password"] = string(c.Password())
```

Comment:
It feels a little fragile to only define these fields and the structure for this part of the API here in the service.  Consider creating a UserPasswordCredential proto message similar to what we do with attributes and then using the handlers.ProtoToStruct method to convert it to the structpb.Struct